bin/gitlab-api-v3

```diff
@@ -103,42 +103,18 @@ try {
 
     exit 0 if !defined $data;
 
-    # Replace all non HASH/ARRAY references with strings.  This is
-    # implemented generically, but really only causes JSON boolean
-    # object to be stringified which makes sense for non-JSON serializers.
-    if ($format ne 'JSON') {
-        $data = clean_data( $data );
```

Comment:
I tried to mention this on the PR summary, but basically `clean_data` forces serialisation of whatever is returned from the API. If all responses are serialised data then this is fine, but since some responses might not be serialisable (those that produce plain text, or binary data), this ends up corrupting the responses.

I imagine the original purpose of that function was to allow for different serialisers to be used (the script actually defaults to YAML).